### PR TITLE
fix: :bug: implement missing feature, edit view comment (fix #8846)

### LINKED ISSUE #8846: Bug / Missing feature, error when editing comment on a view
### Version Information

Server Version: <= v2.11.0-beta.1
CLI Version (for CLI related issue): (unrelated) v2.11.0-beta.1

### Environment

Cloud / OSS / EE

### What is the current behaviour?

Whenever someone tries to edit comment on the view the error message will pop out.
hasura-bug-comment-view

### What is the expected behaviour?

No error message and migration file created when tracking the change via CLI `hasura console`

### How to reproduce the issue?

1. Create a simple view
2. Update/Edit comment via Hasura Console

### Screenshots or Screencast

Error received in the console
hasura-bug-comment-view

Error in postgres
hasura-bug-comment-view-pg-error

### Please provide any traces or logs that could help here.

The generated SQL statement is wrong, the root cause is the reusing of UI table component to edit comments for the view and table, and not passing and checking tableType while doing so.

### Any possible solutions?

Check if the tableType is 'view' if so generate a different query.
I've created a fix, and will link PR, but maybe an additional refactor will be needed after the UI lead understands the issue and decide if reusing the table component is good enough.

### Can you identify the location in the source code where the problem exists?

The missing feature affects all supported dataSources, in my case, it is the postgresql
`graphql-engine/console/src/dataSources/services/postgresql/sqlUtils.ts`
PR should cover all the DS.

### If the bug is confirmed, would you be willing to submit a PR?

Yes, I'm writing this issue to link it to the PR :)

### Keywords

edit view, comment, edit, console, etc.

### PR DESCRIPTION
**Pull request by gnekich**
implement missing feature and fix error when trying to edit view comment

### Description

User will be able to edit comments on views by using Hasura Console UI.

### Changelog

__Component__ : console

__Type__: bugfix/feature (not sure)

__Product__: community-edition

#### Long Changelog

Error message is thrown when user tries to update comment on the view. The root cause is that component for modifying views is using table component which in the end uses getAlterTableCommentSql to generate invalid sql statement. Even if that would be correct naming of the newly generated migration file would be called table_
Wrong migration file with invalid name would be set in saveTableCommentSql(), it would use 'alter_table_' instead of 'alter_view_'

### Related Issues

#8846 

### Solution and Design

Added new function getAlterViewCommentSql based on getAlterTableCommentSql
Also checking tableType variable that was there but not used in saveTableCommentSql

### Steps to test and verify

1.) Create simple view
2.) Open console
3.) Edit comment

### Limitations, known bugs & workarounds

No known limitations.

#### Catalog upgrade

Does this PR change Hasura Catalog version?
- [x] No
- [ ] Yes
  - [ ] Updated docs with SQL for downgrading the catalog 

#### Metadata

Does this PR add a new Metadata feature?
- [x] No
- [ ] Yes
  - Does `run_sql` auto manages the new metadata through schema diffing?
    - [ ] Yes
    - [ ] Not required
  - Does `run_sql` auto manages the definitions of metadata on renaming?
    - [ ] Yes
    - [ ] Not required
  - Does `export_metadata`/`replace_metadata` supports the new metadata added?
    - [ ] Yes
    - [ ] Not required

#### GraphQL
- [x] No new GraphQL schema is generated
- [ ] New GraphQL schema is being generated:
   - [ ] New types and typenames are correlated
   
   

#### Breaking changes

- [x] No Breaking changes
- [ ] There are breaking changes:

  1. Metadata API

     Existing `query` types:
     - [ ] Modify `args` payload which is not backward compatible
     - [ ] Behavioural change of the API
     - [ ] Change in response `JSON` schema
     - [ ] Change in error code
     

  2. GraphQL API

     Schema Generation:
     
     
     - [ ] Change in any `NamedType`
     - [ ] Change in table field names
     

     Schema Resolve:-
     
     - [ ] Change in treatment of `null` value for any input fields 
     

  3. Logging

     - [ ] Log `JSON` schema has changed
     - [ ] Log `type` names have changed

### CODE REVIEW

File: console/src/dataSources/services/postgresql/__tests__/postgresql.test.ts [hunk 1]
```diff
@@ -62,6 +62,21 @@ describe('postgresql datasource tests', () => {
...
+        viewName: 'view_users',
+        schemaName: 'public',
+        comment: "user's comment",
+      });
+      expect(query).toContain('comment on view');
+      expect(query).toContain('"public"."view_users"');
+      expect(query).toContain("E'user\\'s comment'");
+      expect(query).toMatchSnapshot();
+    });
+  });
```
Comment (vijayprasanna13) [inline console/src/dataSources/services/postgresql/__tests__/postgresql.test.ts:78]: ❤️

File: console/src/dataSources/services/mysql/sqlUtils.ts [hunk 2]
```diff
@@ -199,6 +199,15 @@ export const getAlterColumnCommentSql: DataSourcesAPI['getAlterColumnCommentSql'
...
   };
 
+export const getAlterViewCommentSql: DataSourcesAPI['getAlterViewCommentSql'] =
+  ({ viewName, schemaName, comment }) => {
+    const commentStr = sqlEscapeText(comment);
+    return `alter view ${getMySQLNameString(
+      schemaName,
+      viewName
+    )} comment = ${commentStr};`;
+  };
```
Comment (vijayprasanna13) [inline console/src/dataSources/services/mysql/sqlUtils.ts:209]: We can remove this. We don't have support for `mysql_run_sql` query type at the moment.

**Comment by vijayprasanna13** [comment]
@gnekich Thanks for the contribution! 
@ejkkan @lucarestagno Can we add this our review queue?

**Review by ejkkan — APPROVED** [review]
LGMT

Thanks @gnekich

File: console/src/dataSources/services/mysql/sqlUtils.ts (hunk 2 above)
Comment (gnekich) [reply console/src/dataSources/services/mysql/sqlUtils.ts]: @vijayprasanna13 Updated the PR to reflect the requested changes.

**Comment by ejkkan** [comment]
Due to a failed build that triggered the PR to merge for the private repo, this PR was left hanging in its `Open` state.

The functionality has been merged and will be part of our upcoming releases.

Thanks for the contribution, @gnekich

**Pull request closed without merging**